ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -34,6 +36,91 @@ module Winbind = struct
     debug "init_service To be implemented in CP-36085"
 end
 
+let net_cmd = "/usr/bin/net"
+
+let from_config ~name ~err_msg ~config_params =
+  match List.assoc_opt name config_params with
+  | Some v ->
+      v
+  | _ ->
+      raise (Auth_service_error (E_GENERIC, err_msg))
+
+let assert_hostname_valid () =
+  let hostname =
+    (fun __context ->
+      Helpers.get_localhost ~__context |> fun host ->
+      Db.Host.get_hostname ~__context ~self:host)
+    |> Server_helpers.exec_with_new_task "retrieving hostname"
+  in
+  let all_numbers hostname =
```

Comment:
This is fine if it is all we need but these kinds of tests are usually better implemented with regular expression matching as they are more flexible. I'd use the `Re` module for this.

Reply:
Yes, this is what we need and what PBIS currently does,  I would like to keep this behavior for backward compatibility,
but regular expression sounds a better solution and I would like to use it.
btw, For expression  in ocaml, I only found https://ocaml.org/api/Str.html#VALstring_match
Can you give me a link to refer the `Re` module?

Reply:
You can see an example here https://github.com/xapi-project/xen-api/blob/ea27a74a532d2e881f8a88a55319287dc5302e7b/ocaml/xapi/create_misc.ml#L113